### ISSUE TITLE
CORS policy: No 'Access-Control-Allow-Origin'

### ISSUE THREAD

**Issue by FlorianLeMenn** [dunglas/mercure#566]
Trying to setup mercure with cors, but I can't seem to get it to work.

Think it's been like that for a while, since #442 has been closed with wontfix by a stale bot we'd have to keep opening issues? or get some understanding in the issue

I made a POC to use Mercure & Symfony 5 with all configurations from Mercure documentation for Windows : https://github.com/FlorianLeMenn/Symfony-chat-mercure-poc/tree/staging
I try to use Cookie authorization with JWT, but CORS policy fail like that : 

![CORS_error](https://user-images.githubusercontent.com/4291551/136670592-535cb835-8d85-4f56-8d73-24f7f6cd9072.png)

![brave_2021-10-09_20-37-43](https://user-images.githubusercontent.com/4291551/136670601-402fd934-fa19-4b39-a4b1-623dea1b7599.png)

Configuration for mercure.exe : 

![cmd_2021-10-09_20-47-36](https://user-images.githubusercontent.com/4291551/136670668-365f0522-40cb-4718-bcf9-2ee90863fbe0.png)

**Comment by dunglas** [comment]
An you paste the command you use to start your web server? If you are using Symfony CLI, it starts a server on the `127.0.0.1` host, which not the same as `localhost`. So you need to update your Mercure configuration accordingly.

**Comment by dunglas** [comment]
It's `MERCURE_EXTRA_DIRECTIVES`, not `EXTRA_DIRECTIVES`. Can you try that?

**Comment by FlorianLeMenn** [comment]
I try it, same result

![cmd_2021-10-10_15-26-32](https://user-images.githubusercontent.com/4291551/136697760-e675bdf9-89b5-471e-86de-fda5a18cd288.png)

`Access to resource at 'http://localhost:3000/.well-known/mercure?topic=%2Fmessages%2F%7Bid%7D&topic=%2Fping%2F%7Bid%7D' from origin 'http://localhost:8000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

**Comment by dunglas** [comment]
With https://github.com/FlorianLeMenn/Symfony-chat-mercure-poc/pull/1 and the following command (on Unix, I don't have a Windows computer), I don't get any CORS error.

`SERVER_NAME='localhost:3000' MERCURE_PUBLISHER_JWT_KEY='!ChangeMe!' MERCURE_SUBSCRIBER_JWT_KEY='!ChangeMe!' MERCURE_EXTRA_DIRECTIVES='cors_origins http://localhost:8000' ./mercure run`

Be sure to visit `https://localhost:3000` in your browser and to accept the self-signed TLS certificate first.
